src/main/scala/io/iohk/ethereum/db/dataSource/DataSource.scala

```diff
@@ -0,0 +1,26 @@
+package io.iohk.ethereum.db.dataSource
+
+trait DataSource {
+
+  type Key = Array[Byte]
+  type Value = Array[Byte]
+
+  /**
+    * This function obtains the associated value to a key, if there exists one.
+    *
+    * @param namespace which will be searched for the key.
+    * @param key
+    * @return the value associated with the passed key.
+    */
+  def get(namespace: Byte, key: Key): Option[Value]
```

Comment:
This only makes sense if the result is *always* expected to be optional. Which for a DataSource, I don't think we can say that's the case. (or can we?) It's fine to create an operation with the pre condition that a value must exist for the given key. E.g. Map.apply(k: K): V